rfcs/react-components/convergence/slot-children-render-function.md

```diff
@@ -0,0 +1,530 @@
+the merging of the External properties and Internal ones, following given priorities (`props` wins over `defaultProps`).
+
+`getSlots` is invoked at `render` method and will go over every single slot declared on `state.components` and convert every resolved shorthand into a pair of
+element to be render and properties to be passed. In this `getSlots` method, we also ensure that if a given resolved shorthand `children`
+property is a function, than we'll invoke that function:
+
+```ts
+// getSlots
+
+if (typeof children === 'function') {
+  const render = children;
+  // in this case the pair of element to be render and properties to be passed are converted into
+  // a Fragment and {children} where `children` will be the return of invoking the `children` render method
+  return [
+    React.Fragment,
+    {
+      children: render(slot, propsWithoutAsAndChildren),
+    } as unknown as R[K],
+  ];
+}
+```
+
+### Problem 1
+
+The `slot` argument will represent the merging between `state.component.button` value
+(back into the `AccordionHeader` `button` slot example) and `props.button.as` value. `propsWithoutAsAndChildren`
+as the name suggests is all the properties provided by the `resolveShorthand` invocation without including `as` and also `children`
+(since `children` by itself is the render function).
+
+Since we lost the Internal properties on the merging between it and the External ones
+There's no way to provide such the default declared `children` to the render function!
+
+### Problem 2
+
+At `render` method after `getSlots` invocation we have all `slots` and `slotProps` well defined. For the case of wrappers
+we'll also provide an override for `children`, taking precedence over External properties and also Internal properties.
+This will cause us to lose any children provided by a `render` function passed as a children.
+
+### Sum up
+
+At the moment the merging of the 3 instances of properties is done in 3 different steps:
+
+1. `resolveShorthand` will merge External properties with Internal properties (external takes precedence)
+2. `getSlots` will **wrongly** filter properties on case of render function as `children`
+3. on slot rendering of the `render` method the merged properties provided by
+   `getSlots` will be merged with the Overrides (overrides takes precedence),
+   once again overriding `children` on the case of `wrappers`
+
+   ```tsx
+      // the third step is the render of the slot itself
+      <slots.button {...slotProps.button}>
+   ```
+
+Properties will be lost after those steps, mainly `children` will be lost, affecting the render function mechanism.
+
+Merging the properties is a complex scenario that should not be splitted into multiple steps as all properties
+must be considered when merging them, since we got peculiar scenarios involving `children` mostly
+
+## Detailed Design or Proposal
+
+### Option A: Custom JSX Pragma
```

Comment:
Should  this RFC contain more options ? I dont see any "Option B" etc.  if it's only about Custom pragma can we remove that "option" to not confuse reader?

Reply:
There are other options, this is not the only possible solution (but it's definitely the most elegant one!)

I'll see to add an alternative option just for the sake of better reasoning